docs/migration-guide.md

```diff
@@ -0,0 +1,10 @@
+# Migration Guide
+- [Upgrading from Microsoft.Spark 0.x to 1.0](#upgrading-from-microsoft.spark-0.x-to-1.0)
```

Comment:
Do we need this to be a self-referencing link?

Reply:
Or is it there coz we might add more sections to this doc

Reply:
We should update this file every time we need new migration guide steps between versions.  The table of contents is there so we can add more sections later.